### PR TITLE
notebooks: log additional events and enable stories snapshotting

### PR DESCRIPTION
**Pull request by novoselrok**
Adds the majority of events listed in https://github.com/sourcegraph/sourcegraph/issues/30087.

Also enables snapshotting for two notebook stories.

### CODE REVIEW

File: client/web/src/search/notebook/SearchNotebook.story.tsx [hunk 1]
```diff
@@ -17,9 +17,9 @@ import { SearchNotebook } from './SearchNotebook'
 
 import { BlockInit } from '.'
 
-const { add } = storiesOf('web/search/notebook/SearchNotebook', module).addDecorator(story => (
-    <div className="p-3 container">{story()}</div>
-))
+const { add } = storiesOf('web/search/notebook/SearchNotebook', module)
+    .addDecorator(story => <div className="p-3 container">{story()}</div>)
+    .addParameters({ chromatic: { disableSnapshots: false } })
```
Comment (erzhtor) [inline client/web/src/search/notebook/SearchNotebook.story.tsx:22]: Would be nice to migrate to CSF.

**Review by erzhtor — APPROVED** [review]
LGTM!

**Comment by limitedmage** [comment]
@novoselrok Does this resolve all the notebooks-related storybook work for #30336?

**Comment by novoselrok** [comment]
@limitedmage Yep, if I understood correctly, I had to enable the notebook stories we want to keep screenshotting (only two seemed to be worth it).

**Pull request merged**